### PR TITLE
Fix/opening times abutting midnight

### PR DESCRIPTION
**Pull request by markysoft**
Updated to handle times spanning midnight and 24 hours opening pharmacies

### CODE REVIEW

**Comment by neilbmclaughlin** [comment]
[Update handling of opening times to support those that are open over Midnight](https://trello.com/c/tSfK3VzK/656-update-handling-of-opening-times-to-support-those-that-are-open-over-midnight)

**Comment by neilbmclaughlin** [comment]
Test Comment

**Comment by neilbmclaughlin** [comment]
Test Comment

**Comment by neilbmclaughlin** [comment]
test

**Comment by neilbmclaughlin** [comment]
@markysoft - could you rebase please

File: app/lib/midnightSpanCorrector.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const utils = require('../lib/utils');
+
+const minimunCloseTimeMins = 1;
```
Comment (neilbmclaughlin) [inline app/lib/midnightSpanCorrector.js]: Typo

File: test/unit/lib/midnightSpanCorrector.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    moment.add(week, 'week');
+  }
+  return moment;
+}
+
+function momentsShouldBeSame(moment1, moment2) {
+  expect(moment1).to.be.sameMoment(moment2);
+}
+
+describe('Midnight Span Corrector', () => {
```
Comment (neilbmclaughlin) [inline test/unit/lib/midnightSpanCorrector.js]: Do we need any tests to show that alternate hours which span midnight are handled?

**Comment by markysoft** [comment]
Typo fixed

**Comment by neilbmclaughlin** [comment]
Had to hand roll a [stack](https://rancher.nhschoices.net/env/1a7/apps/stacks/1st271) in Rancher as the CI scripts don't yet exist in this branch.

The app is available [here](http://nearby-services-api-pr-34.dev.c2s.nhschoices.net/finders/results?location=RH6+0NN&context=)

DATETIME is set for 'Thu 16 Feb 2017 03:02:34'. It was set by adding an environment variable in Rancher to the nearby-services-api service via 'upgrade'. Had to do this via UI but I'm guessing it should be possible to do it from the command line.

`rancher inspect nearby-services-api-pr-34/nearby-services-api | jq '.launchConfig.environment'`

**Pull request merged**